docs/dev/run_the_etl.rst

```diff
@@ -11,24 +11,182 @@ just want to use already processed data.
+tables.
+
+SDAs are created by applying the ``@asset`` decorator to a function.
+
+**IO Managers:**
+
+    *IO Managers are user-provided objects that store asset outputs
+    and load them as inputs to downstream assets.*
+
+Each asset has an `IO Manager  <https://docs.dagster.io/concepts/io-management/io-managers>`__
+that tells Dagster how to handle the objects returned by the software
+defined asset's underlying function. The IO Managers in PUDL read and
+write dataframes to and from databases and parquet files.
+
+**Resources:**
+
+   *Resources are objects that are shared across the implementations of
+   multiple software-defined assets and that can be plugged in
+   after defining those assets.*
+
+`Resources <https://docs.dagster.io/concepts/resources>`__ are helpful
+for sharing python objects amongst assets.
+For example, PUDL uses a Datastore resource that multiple assets use
+to pull data from PUDL's raw data archives on Zenodo.
+
+Generally, inputs to assets should either be other assets or
+python objects in Resources.
+
+**Jobs**:
+`Jobs <https://docs.dagster.io/concepts/ops-jobs-graphs/jobs>`__
+are preconfigured collections of assets, resources and IO Managers.
+Jobs are the main unit of execution in Dagster.
+
+**Definitions**:
+`Definitions  <https://docs.dagster.io/concepts/code-locations>`__
+are collections of assets, resources, IO managers and jobs that can
+be loaded into dagit and executed.
+
+There are two main Definitions in the PUDL processing pipeline:
+
+1. ``pudl.ferc_to_sqlite`` :doc:`converts the FERC Form 1 DBF/XBRL files <clone_ferc1>`
+   into a single large `SQLite <https://sqlite.org>`__ database so that the data is
+   easier to extract, and so all of the raw FERC Form 1 data is available in a modern
+   format. You must run this script before you can run ``pudl_etl``.
+2. ``pudl.etl`` coordinates the "Extract, Transform, Load" process that processes
    20+ years worth of data from the FERC Form 1 database, dozens of EIA spreadsheets,
    and the thousands of CSV files that make up the EPA CEMS hourly emissions data into
    a clean, well normalized SQLite database (for the FERC and EIA data), and an `Apache
    Parquet <https://parquet.apache.org/>`__ dataset that is partitioned by state and
    year (for the EPA CEMS).
 
+Both definitions have two preconfigured jobs:
+  - ``etl_fast`` processes one year of data
+  - ``etl_full`` processes all years of data
+
+.. _run-dagit:
+
+Running the ETL with Dagit
+--------------------------
+Dagster needs a directory to store run logs and some assets. Create a new directory
+outside of the pudl respository directory called ``dagster_home/``. Then set the
+``DAGSTER_HOME`` variable to the path of the new directory:
+
+.. code-block:: console
+
+    mamba env config vars set DAGSTER_HOME=</path/to/dagster_home/directory/>
+    mamba activate pudl-dev
+
+Once ``DAGSTER_HOME`` is set, launch Dagit by running:
+
+.. code-block::
+
+    make dagit
```

Comment:
When I run `make dagit` I get a bunch of errors, but they don't seem to prevent dagit from starting up:

<details><summary>stack traces</summary>

```
dagit -m pudl.etl -m pudl.ferc_to_sqlite
2023-03-05 11:04:21 -0600 - dagster - WARNING - No dagster instance configuration file (dagster.yaml) found at /Users/zane/code/catalyst/dagster-home. Defaulting to loading and storing all metadata with /Users/zane/code/catalyst/dagster-home. If this is the desired behavior, create an empty dagster.yaml file in /Users/zane/code/catalyst/dagster-home.
Traceback (most recent call last):
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/__main__.py", line 1, in <module>
    from ._cli import main
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_cli/__init__.py", line 4, in <module>
    from .api import api_cli
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_cli/api.py", line 14, in <module>
    from dagster._cli.workspace.cli_target import (
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_cli/workspace/__init__.py", line 1, in <module>
    from .cli_target import (
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_cli/workspace/cli_target.py", line 25, in <module>
    from dagster._core.host_representation.external import ExternalRepository
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_core/host_representation/__init__.py", line 55, in <module>
    from .repository_location import (
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_core/host_representation/repository_location.py", line 10, in <module>
    from dagster._api.list_repositories import sync_list_repositories_grpc
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_api/list_repositories.py", line 6, in <module>
    from dagster._grpc.types import ListRepositoriesResponse
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/__init__.py", line 11, in <module>
    from .client import (
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/client.py", line 9, in <module>
    import grpc
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/grpc/__init__.py", line 22, in <module>
    from grpc import _compression
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/grpc/_compression.py", line 15, in <module>
    from grpc._cython import cygrpc
ImportError: dlopen(/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/grpc/_cython/cygrpc.cpython-310-darwin.so, 0x0002): symbol not found in flat namespace '_CFRelease'
/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_core/workspace/context.py:602: UserWarning: Error loading repository location pudl.etl:Exception: gRPC server exited with return code 1 while starting up with the command: "/Users/zane/mambaforge/envs/pudl-dev/bin/python -m dagster api grpc --lazy-load-user-code --socket /var/folders/ps/6jyqvztj5fq_tvhwx5h59d7w0000z8/T/tmpem1k3swl --heartbeat --heartbeat-timeout 45 --fixed-server-id aa9fef1a-ff6c-4ef7-bf63-d5245d55945f --log-level info --inject-env-vars-from-instance --instance-ref {"__class__": "InstanceRef", "compute_logs_data": {"__class__": "ConfigurableClassData", "class_name": "LocalComputeLogManager", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home/storage\n", "module_name": "dagster.core.storage.local_compute_log_manager"}, "custom_instance_class_data": null, "event_storage_data": {"__class__": "ConfigurableClassData", "class_name": "SqliteEventLogStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home/history/runs/\n", "module_name": "dagster.core.storage.event_log"}, "local_artifact_storage_data": {"__class__": "ConfigurableClassData", "class_name": "LocalArtifactStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home\n", "module_name": "dagster.core.storage.root"}, "run_coordinator_data": {"__class__": "ConfigurableClassData", "class_name": "DefaultRunCoordinator", "config_yaml": "{}\n", "module_name": "dagster.core.run_coordinator"}, "run_launcher_data": {"__class__": "ConfigurableClassData", "class_name": "DefaultRunLauncher", "config_yaml": "{}\n", "module_name": "dagster"}, "run_storage_data": {"__class__": "ConfigurableClassData", "class_name": "SqliteRunStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home/history/\n", "module_name": "dagster.core.storage.runs"}, "schedule_storage_data": {"__class__": "ConfigurableClassData", "class_name": "SqliteScheduleStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home/schedules\n", "module_name": "dagster.core.storage.schedules"}, "scheduler_data": {"__class__": "ConfigurableClassData", "class_name": "DagsterDaemonScheduler", "config_yaml": "{}\n", "module_name": "dagster.core.scheduler"}, "secrets_loader_data": null, "settings": {}, "storage_data": {"__class__": "ConfigurableClassData", "class_name": "DagsterSqliteStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home\n", "module_name": "dagster.core.storage.sqlite_storage"}} --location-name pudl.etl -m pudl.etl -d /Users/zane/code/catalyst/pudl"

Stack Trace:
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_core/host_representation/grpc_server_registry.py", line 208, in _get_grpc_endpoint
    server_process = GrpcServerProcess(
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/server.py", line 1243, in __init__
    server_process = open_server_process(
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/server.py", line 1134, in open_server_process
    wait_for_grpc_server(server_process, client, subprocess_args, timeout=startup_timeout)
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/server.py", line 1068, in wait_for_grpc_server
    raise Exception(

  warnings.warn(
Traceback (most recent call last):
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/runpy.py", line 196, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/__main__.py", line 1, in <module>
    from ._cli import main
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_cli/__init__.py", line 4, in <module>
    from .api import api_cli
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_cli/api.py", line 14, in <module>
    from dagster._cli.workspace.cli_target import (
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_cli/workspace/__init__.py", line 1, in <module>
    from .cli_target import (
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_cli/workspace/cli_target.py", line 25, in <module>
    from dagster._core.host_representation.external import ExternalRepository
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_core/host_representation/__init__.py", line 55, in <module>
    from .repository_location import (
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_core/host_representation/repository_location.py", line 10, in <module>
    from dagster._api.list_repositories import sync_list_repositories_grpc
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_api/list_repositories.py", line 6, in <module>
    from dagster._grpc.types import ListRepositoriesResponse
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/__init__.py", line 11, in <module>
    from .client import (
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/client.py", line 9, in <module>
    import grpc
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/grpc/__init__.py", line 22, in <module>
    from grpc import _compression
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/grpc/_compression.py", line 15, in <module>
    from grpc._cython import cygrpc
ImportError: dlopen(/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/grpc/_cython/cygrpc.cpython-310-darwin.so, 0x0002): symbol not found in flat namespace '_CFRelease'
/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_core/workspace/context.py:602: UserWarning: Error loading repository location pudl.ferc_to_sqlite:Exception: gRPC server exited with return code 1 while starting up with the command: "/Users/zane/mambaforge/envs/pudl-dev/bin/python -m dagster api grpc --lazy-load-user-code --socket /var/folders/ps/6jyqvztj5fq_tvhwx5h59d7w0000z8/T/tmpan_m4k7j --heartbeat --heartbeat-timeout 45 --fixed-server-id 61b86fed-1397-48e4-be6d-cd9383de2fae --log-level info --inject-env-vars-from-instance --instance-ref {"__class__": "InstanceRef", "compute_logs_data": {"__class__": "ConfigurableClassData", "class_name": "LocalComputeLogManager", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home/storage\n", "module_name": "dagster.core.storage.local_compute_log_manager"}, "custom_instance_class_data": null, "event_storage_data": {"__class__": "ConfigurableClassData", "class_name": "SqliteEventLogStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home/history/runs/\n", "module_name": "dagster.core.storage.event_log"}, "local_artifact_storage_data": {"__class__": "ConfigurableClassData", "class_name": "LocalArtifactStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home\n", "module_name": "dagster.core.storage.root"}, "run_coordinator_data": {"__class__": "ConfigurableClassData", "class_name": "DefaultRunCoordinator", "config_yaml": "{}\n", "module_name": "dagster.core.run_coordinator"}, "run_launcher_data": {"__class__": "ConfigurableClassData", "class_name": "DefaultRunLauncher", "config_yaml": "{}\n", "module_name": "dagster"}, "run_storage_data": {"__class__": "ConfigurableClassData", "class_name": "SqliteRunStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home/history/\n", "module_name": "dagster.core.storage.runs"}, "schedule_storage_data": {"__class__": "ConfigurableClassData", "class_name": "SqliteScheduleStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home/schedules\n", "module_name": "dagster.core.storage.schedules"}, "scheduler_data": {"__class__": "ConfigurableClassData", "class_name": "DagsterDaemonScheduler", "config_yaml": "{}\n", "module_name": "dagster.core.scheduler"}, "secrets_loader_data": null, "settings": {}, "storage_data": {"__class__": "ConfigurableClassData", "class_name": "DagsterSqliteStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home\n", "module_name": "dagster.core.storage.sqlite_storage"}} --location-name pudl.ferc_to_sqlite -m pudl.ferc_to_sqlite -d /Users/zane/code/catalyst/pudl"

Stack Trace:
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_core/host_representation/grpc_server_registry.py", line 208, in _get_grpc_endpoint
    server_process = GrpcServerProcess(
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/server.py", line 1243, in __init__
    server_process = open_server_process(
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/server.py", line 1134, in open_server_process
    wait_for_grpc_server(server_process, client, subprocess_args, timeout=startup_timeout)
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/server.py", line 1068, in wait_for_grpc_server
    raise Exception(

  warnings.warn(
2023-03-05 11:04:22 -0600 - dagit - INFO - Serving dagit on http://127.0.0.1:3000 in process 81488
```

</details>

Reply:
When I go to the `dagit` interface, I find that both of the code locations have errors, and they appear to be different from the errors that showed up in the stack traces above, but maybe related. Lots of stuff about `grpc`

Reply:
Unfortunately couldn't test out the parts below here.

Reply:
Strange :/ I wasn't able to reproduce the grpc errors. What were the errors you got in dagit?

Reply:
<details><summary>pudl.etl errors</summary>

```
Exception: gRPC server exited with return code 1 while starting up with the command: "/Users/zane/mambaforge/envs/pudl-dev/bin/python -m dagster api grpc --lazy-load-user-code --socket /var/folders/ps/6jyqvztj5fq_tvhwx5h59d7w0000z8/T/tmp6llo54v4 --heartbeat --heartbeat-timeout 45 --fixed-server-id f162f722-a90b-41e8-9c78-5e7f7af3ab51 --log-level info --inject-env-vars-from-instance --instance-ref {"__class__": "InstanceRef", "compute_logs_data": {"__class__": "ConfigurableClassData", "class_name": "LocalComputeLogManager", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home/storage\n", "module_name": "dagster.core.storage.local_compute_log_manager"}, "custom_instance_class_data": null, "event_storage_data": {"__class__": "ConfigurableClassData", "class_name": "SqliteEventLogStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home/history/runs/\n", "module_name": "dagster.core.storage.event_log"}, "local_artifact_storage_data": {"__class__": "ConfigurableClassData", "class_name": "LocalArtifactStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home\n", "module_name": "dagster.core.storage.root"}, "run_coordinator_data": {"__class__": "ConfigurableClassData", "class_name": "DefaultRunCoordinator", "config_yaml": "{}\n", "module_name": "dagster.core.run_coordinator"}, "run_launcher_data": {"__class__": "ConfigurableClassData", "class_name": "DefaultRunLauncher", "config_yaml": "{}\n", "module_name": "dagster"}, "run_storage_data": {"__class__": "ConfigurableClassData", "class_name": "SqliteRunStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home/history/\n", "module_name": "dagster.core.storage.runs"}, "schedule_storage_data": {"__class__": "ConfigurableClassData", "class_name": "SqliteScheduleStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home/schedules\n", "module_name": "dagster.core.storage.schedules"}, "scheduler_data": {"__class__": "ConfigurableClassData", "class_name": "DagsterDaemonScheduler", "config_yaml": "{}\n", "module_name": "dagster.core.scheduler"}, "secrets_loader_data": null, "settings": {}, "storage_data": {"__class__": "ConfigurableClassData", "class_name": "DagsterSqliteStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home\n", "module_name": "dagster.core.storage.sqlite_storage"}} --location-name pudl.etl -m pudl.etl -d /Users/zane/code/catalyst/pudl"
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_core/host_representation/grpc_server_registry.py", line 208, in _get_grpc_endpoint
    server_process = GrpcServerProcess(
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/server.py", line 1243, in __init__
    server_process = open_server_process(
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/server.py", line 1134, in open_server_process
    wait_for_grpc_server(server_process, client, subprocess_args, timeout=startup_timeout)
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/server.py", line 1068, in wait_for_grpc_server
    raise Exception(
```

</details>

<details><summary>pudl.ferc_to_sqlite errors</summary>

```
Exception: gRPC server exited with return code 1 while starting up with the command: "/Users/zane/mambaforge/envs/pudl-dev/bin/python -m dagster api grpc --lazy-load-user-code --socket /var/folders/ps/6jyqvztj5fq_tvhwx5h59d7w0000z8/T/tmp33sjbtki --heartbeat --heartbeat-timeout 45 --fixed-server-id 8fea512e-81b2-426f-88e4-4a5bf6f238c2 --log-level info --inject-env-vars-from-instance --instance-ref {"__class__": "InstanceRef", "compute_logs_data": {"__class__": "ConfigurableClassData", "class_name": "LocalComputeLogManager", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home/storage\n", "module_name": "dagster.core.storage.local_compute_log_manager"}, "custom_instance_class_data": null, "event_storage_data": {"__class__": "ConfigurableClassData", "class_name": "SqliteEventLogStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home/history/runs/\n", "module_name": "dagster.core.storage.event_log"}, "local_artifact_storage_data": {"__class__": "ConfigurableClassData", "class_name": "LocalArtifactStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home\n", "module_name": "dagster.core.storage.root"}, "run_coordinator_data": {"__class__": "ConfigurableClassData", "class_name": "DefaultRunCoordinator", "config_yaml": "{}\n", "module_name": "dagster.core.run_coordinator"}, "run_launcher_data": {"__class__": "ConfigurableClassData", "class_name": "DefaultRunLauncher", "config_yaml": "{}\n", "module_name": "dagster"}, "run_storage_data": {"__class__": "ConfigurableClassData", "class_name": "SqliteRunStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home/history/\n", "module_name": "dagster.core.storage.runs"}, "schedule_storage_data": {"__class__": "ConfigurableClassData", "class_name": "SqliteScheduleStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home/schedules\n", "module_name": "dagster.core.storage.schedules"}, "scheduler_data": {"__class__": "ConfigurableClassData", "class_name": "DagsterDaemonScheduler", "config_yaml": "{}\n", "module_name": "dagster.core.scheduler"}, "secrets_loader_data": null, "settings": {}, "storage_data": {"__class__": "ConfigurableClassData", "class_name": "DagsterSqliteStorage", "config_yaml": "base_dir: /Users/zane/code/catalyst/dagster-home\n", "module_name": "dagster.core.storage.sqlite_storage"}} --location-name pudl.ferc_to_sqlite -m pudl.ferc_to_sqlite -d /Users/zane/code/catalyst/pudl"
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_core/host_representation/grpc_server_registry.py", line 208, in _get_grpc_endpoint
    server_process = GrpcServerProcess(
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/server.py", line 1243, in __init__
    server_process = open_server_process(
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/server.py", line 1134, in open_server_process
    wait_for_grpc_server(server_process, client, subprocess_args, timeout=startup_timeout)
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/server.py", line 1068, in wait_for_grpc_server
    raise Exception(
```

</details>

Reply:
Looking in my `DAGSTER_HOME` it did create some files & directories:
```
├── history
│   ├── runs
│   │   └── index.db
│   └── runs.db
└── schedules
    └── schedules.db
```

Reply:
Installed packages:
```
grpcio                           1.47.5
grpcio-health-checking           1.43.0
```

Reply:
Hmm I'm working with the same versions. I'll do some research.

Reply:
@jdangerx @zschira or @katie-lamb did you run into this error when setting up dagster?

Reply:
I get pretty similar errors with the basic test suggested in [this issue](https://github.com/dagster-io/dagster/issues/4289)

My guess is there's something wrong with some dependency / cython or C extension, given where it ultimately barfs. Maybe it's an Apple Silicon issue?

```bash
dagster api grpc -p 4000 -f hello_world.py         
```

```                                                                                                                               
Traceback (most recent call last):
  File "/Users/zane/mambaforge/envs/pudl-dev/bin/dagster", line 5, in <module>
    from dagster.cli import main
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_module_alias_map.py", line 72, in exec_module
    base_module = importlib.import_module(self.base_spec.name)
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_cli/__init__.py", line 4, in <module>
    from .api import api_cli
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_cli/api.py", line 14, in <module>
    from dagster._cli.workspace.cli_target import (
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_cli/workspace/__init__.py", line 1, in <module>
    from .cli_target import (
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_cli/workspace/cli_target.py", line 25, in <module>
    from dagster._core.host_representation.external import ExternalRepository
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_core/host_representation/__init__.py", line 55, in <module>
    from .repository_location import (
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_core/host_representation/repository_location.py", line 10, in <module>
    from dagster._api.list_repositories import sync_list_repositories_grpc
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_api/list_repositories.py", line 6, in <module>
    from dagster._grpc.types import ListRepositoriesResponse
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/__init__.py", line 11, in <module>
    from .client import (
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/dagster/_grpc/client.py", line 9, in <module>
    import grpc
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/grpc/__init__.py", line 22, in <module>
    from grpc import _compression
  File "/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/grpc/_compression.py", line 15, in <module>
    from grpc._cython import cygrpc
ImportError: dlopen(/Users/zane/mambaforge/envs/pudl-dev/lib/python3.10/site-packages/grpc/_cython/cygrpc.cpython-310-darwin.so, 0x0002): symbol not found in flat namespace '_CFRelease'

Reply:
It definitely seems to be an issue with the version of `grpcio`.  I have tried installing every minor version from 1.47 to 1.51 with `pip` and none of them seemed to work. They also all had to build wheels locally.

If I install the most recent version (1.51) using `mamba` then I can run `dagster -m pudl.etl` just fine, and browse around in `dagit` without seeing any errors.  Unfortunately, Dagster requires `grpcio<1.48.1` so even though this seems to work it doesn't solve the problem.  It also appears that many versions of `grpcio` that are available on PyPI are not available via `conda-forge`, so I wasn't able to install 1.48.0 using mamba to see if it worked. It's available for `osx-arm64` under python 3.9 but not 3.10.  1.48.1 is available for Python 10, but the last version that Dagster supports is 1.48.0.

You're using an M1 mac, and 1.47.5 is getting installed by `pip` when you create your environment, and it ends up working for you?

Reply:
@bendnorman Thus far none of the [solutions the dagster folks have suggested](https://github.com/dagster-io/dagster/issues/10262#issuecomment-1460337891) has worked. I've added `grpcio==1.46.1` to the `environment.yml` file and that works. Not sure how this will affect anything happening under Tox, but it does let me play with dagit. Should I push that change to this PR?

Reply:
Bummer :/ Have you tried testing the change with tox?